beacon-chain/forkchoice/doubly-linked-tree/forkchoice.go

```diff
@@ -714,3 +714,80 @@ func (f *ForkChoice) ParentRoot(root [32]byte) ([32]byte, error) {
 	}
 	return n.parent.root, nil
 }
+
+// UpdateVotesOnPayloadAttestation processes a new aggregated
+// payload attestation message and updates
+// the Payload Timeliness Committee (PTC) votes for the corresponding block.
+// It may update payload boost roots based on the attestation data.
+func (s *Store) UpdateVotesOnPayloadAttestation(
+	payloadAttestation *ethpb.PayloadAttestation,
+	isFromBlock bool) error {
+	// Extract the attestation data and convert the beacon block root to a 32-byte array
+	data := payloadAttestation.Data
+	blockRoot := bytesutil.ToBytes32(data.BeaconBlockRoot)
+
+	// Check if the block exists in the store
+	node, ok := s.nodeByRoot[blockRoot]
+	if !ok {
+		return errors.New("beacon block root unknown")
+	}
+
+	if isFromBlock {
```

Comment:
In forkchoice we update the ptc vote independent of the timing of the block, but only update the boosts if the block is early.

Reply:
fixed in my next commit